src/crystal/system/win32/process.cr

```diff
@@ -190,6 +186,25 @@ struct Crystal::System::Process
   def self.chroot(path)
     raise NotImplementedError.new("Process.chroot")
   end
+
+  protected def self.make_env_block(env, clear_env : Bool) : UInt16*
+    return Pointer(UInt16).null if !env && !clear_env
+
+    final_env = {} of String => {String, String}
+    unless clear_env
+      Crystal::System::Env.each do |key, val|
+        final_env[key.upcase] = {key, val}
+      end
+    end
+    env.try &.each do |(key, val)|
+      if val
+        final_env[key.upcase] = {key, val}
+      else
+        final_env.delete key.upcase
+      end
+    end
+    Crystal::System::Env.make_env_block(final_env.each_value)
```

Comment:
The collection is `{key.upcase, {key, value}}`, the values are `{key, value}`, sooo yeah

Reply:
This means an additional allocation for the iterator which does not seem to be strictly necessary. Since this is all internal implementation, the interfaces don't need to be polished. 
Maybe `Env.make_env_block` implementation should actually be here in `Process` because it is only used here and doesn't have anything in common with the other methods in `Env` which build on the win32 API (`GetEnvironmentVariableW`/`SetEnvironmentVariableW`).

Reply:
Well no, it certainly does have something in common, as the method `each` parses the exact kind of structure that `make_env_block` produces.

Reply:
But you are correct that the implementation is not actually shared and the iterator is not strictly necessary.

Reply:
@oprypin It could be just `final_env.values`, no?

Reply:
That still means another allocation. IMO It would be perfectly fine to just pass `final_env`. It just needs to work.

Reply:
I dont mind merging this as-is.